pkgs/development/tools/literate-programming/Literate/default.nix

```diff
@@ -1,28 +1,48 @@
-{ lib, stdenv, fetchFromGitHub, ldc, dub }:
+{
+  lib,
+  buildDubPackage,
+  fetchFromGitHub,
+}:
 
-stdenv.mkDerivation {
+buildDubPackage {
   pname = "Literate";
   version = "unstable-2021-01-22";
 
   src = fetchFromGitHub {
     owner = "zyedidia";
     repo = "Literate";
     rev = "7004dffec0cff3068828514eca72172274fd3f7d";
-    sha256 = "sha256-erNFe0+FlrslEENyO/YxYQbmec0voK31UWr5qVt+nXQ=";
+    hash = "sha256-erNFe0+FlrslEENyO/YxYQbmec0voK31UWr5qVt+nXQ=";
     fetchSubmodules = true;
   };
 
-  buildInputs = [ ldc dub ];
+  # as there aren't any non-local dub dependencies, this file just has any empty list
+  dubDeps = ./deps.nix;
 
-  HOME = "home";
+  # generate the actual .d source files defined in .lit files
+  preBuild = ''
+    make d-files
+  '';
 
-  installPhase = "install -D bin/lit $out/bin/lit";
+  doCheck = true;
 
-  meta = with lib; {
+  checkPhase = ''
+    runHook preCheck
+    dub test
+    runHook postCheck
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 bin/lit -t $out/bin
+    runHook preInstall
+  '';
```

Comment:
Is there no standardised install command for dub?

Reply:
No, but I can file an issue asking about it. I feel like such a feature would be useful to me personally anyway.

Reply:
There exist an optional `targetPath` value inside `dub.sdl` or `dub.json`, AFAICT defaulting to `./.`. It's where it puts the final binaries and the files inside `copyPath`.

Perhaps some `jq` based manipulation could be done to the file to add/update it and set it to a fixed directory, though I am not certain if this is robust enough.
(You can convert from `.sdl` to `.json` with a `dub` command, so `jq` could be used in either case)

Reply:
alternatively this could be an argument to buildDubPackage